supervisor/addons/validate.py

```diff
@@ -455,6 +456,7 @@ def _migrate(config: dict[str, Any]):
     {
         vol.Required(ATTR_NAME): str,
         vol.Optional(ATTR_DESCRIPTON): vol.Maybe(str),
+        vol.Optional(ATTR_FIELDS): {str: vol.Self},
```

Comment:
This is quite nice. I honestly didn't realize `vol.Self` was an option for a making a recursive schema